examples/map/map_segment.py

```diff
@@ -0,0 +1,98 @@
+"""
+=======================================================
+Segmenting a Map based on transformation of coordinates
+=======================================================
+This example demonstrates extracting a region of a particular map based on
+world coordinates in different systems.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import SkyCoord
+
+import sunpy.map
+from sunpy.coordinates import NorthOffsetFrame
+from sunpy.data.sample import AIA_171_IMAGE
+
+######################################################################
+# We start with the sample data.
+smap = sunpy.map.Map(AIA_171_IMAGE)
+
+######################################################################
+# A utility function gives us access to the helioprojective coordinate of each
+# pixel in this map. From this we then transform the coordinates to the required frame.
+# For this example we are going to extract a region based on the
+# heliographic Stonyhurst coordinates, so we transform to that frame.
+all_hpc = sunpy.map.all_coordinates_from_map(smap)
+all_hgs = all_hpc.transform_to("heliographic_stonyhurst")
+
+######################################################################
+# Let's then segment the data based on coordinates and create a boolean mask
+# where `True` indicates invalid or deselected data.
+# Numpy's masked arrays allow for a combination of standard numpy array and
+# a mask array. When an element of the mask is `True`, the corresponding element
+# of the associated array is said to be masked (invalid).
+# For more information about numpy's masked arrays see :mod:`numpy.ma`.
+# We now mask out all values not in our coordinate range and where the coordinates are NaN.
```

Comment:
```suggestion
# We now mask out all values not in our coordinate range or where the
# coordinates are NaN (because they could not be transformed to the
# surface of the Sun).
```